File: Clases 2018/Clases-C#/Simulacro_parcial_2_2018 (creacion de bug e historico en transaccion)/C#_TPS_BugTracker/TPS_BugTracker/DataLayer/DAOs/CriticidadDao.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TPS_BugTracker.BusinessLayer.Entities;

namespace TPS_BugTracker.DataLayer.DAOs
{
    class CriticidadDao
    {
        public List<Criticidad> getAll()
        {
            List<Criticidad> lst = new List<Criticidad>();
            DataManager dm = new DataManager();

            try
            {
                dm.Open();

                foreach (DataRow row in dm.ConsultaSQL("Select * from Criticidades").Rows)
                    lst.Add(map(row));
            }
            catch (Exception ex)
            {
                dm.Rollback();
            }
            finally
            {
                // Cierra la conexión 
                dm.Close();
            }

            return lst;
        }

        private Criticidad map(DataRow row)
        {
            Criticidad oCriticidad = new Criticidad();
            oCriticidad.id_criticidad = Convert.ToInt32(row["id_criticidad"].ToString());
            oCriticidad.nombre = row["n_criticidad"].ToString();
            return oCriticidad;
        }
    }
}

File: Clases 2018/Clases-C#/Simulacro_parcial_2_2018 (creacion de bug e historico en transaccion)/VB_TPS_BugTracker/TPS_BugTracker/DataLayer/DAOs/CriticidadDao.vb
Class CriticidadDao
    Public Function getAll() As List(Of Criticidad)
        Dim lst As List(Of Criticidad) = New List(Of Criticidad)()
        Dim dm As DataManager = New DataManager()

        Try
            dm.Open()

            For Each row As DataRow In dm.ConsultaSQL("Select * from Criticidades").Rows
                lst.Add(map(row))
            Next

        Catch ex As Exception
            dm.Rollback()
        Finally
            dm.Close()
        End Try

        Return lst
    End Function

    Private Function map(ByVal row As DataRow) As Criticidad
        Dim oCriticidad As Criticidad = New Criticidad()
        oCriticidad.id_criticidad = Convert.ToInt32(row("id_criticidad").ToString())
        oCriticidad.nombre = row("n_criticidad").ToString()
        Return oCriticidad
    End Function
End Class